src/modal/modal-stack.ts

```diff
@@ -45,6 +46,7 @@ export class NgbModalStack {
     let windowCmptRef: ComponentRef<NgbModalWindow> = this._attachWindowComponent(containerEl, contentRef);
     let ngbModalRef: NgbModalRef = new NgbModalRef(windowCmptRef, contentRef, backdropCmptRef, options.beforeDismiss);
 
+    ngbModalRef.result.then(revertPaddingForScrollBar, revertPaddingForScrollBar);
```

Comment:
Maybe write as `ngbModalRef.result.finally(revertPaddingForScrollBar)` ?

Reply:
Forget it, it seems like browser support is still lacking... https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally